### ISSUE TITLE
No Analytics being tracked in iOS

### ISSUE THREAD

**Issue by zanechua** [microsoft/appcenter-sdk-react-native#584]
### **Description**

Unable to get any analytics tracking information from iOS builds of an Expo ejected react native application.

### **Repro Steps**

Please list the steps used to reproduce your issue.

1. Follow the documentation on setting up AppCenter (https://docs.microsoft.com/en-us/appcenter/sdk/getting-started/react-native)
2. Build IPA and deploy to a development device
3. Launch the app

### **Details**

I have tried both the integration with and without react-native link and neither work. Both builds compile fine with the fixes available in the current repo (Specifically for the header search paths for the integration with react-native link)

1. Which version of the App Center SDK are you using?
    - 2.0.1
2. Which OS version did you experience the issue on?
    - iOS 12.2
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    - iPhone XR Simulator
    - iPhone XR
4. What third party libraries are you using?
    - no ios third party libraries except the ones that come with an Expo ejected app.
    - Podfile (https://gist.github.com/zanechua/a326b4165d16e7aa07ea7cccc28bd900)
5. Run the following command and paste the output below: `react-native info`
```
  React Native Environment Info:
    System:
      OS: macOS 10.14.4
      CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
      Memory: 55.68 MB / 8.00 GB
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 10.15.3 - /usr/local/bin/node
      Yarn: 1.13.0 - /usr/local/bin/yarn
      npm: 6.4.1 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.2, macOS 10.14, tvOS 12.2, watchOS 5.2
      Android SDK:
        API Levels: 23, 25, 26, 27, 28
        Build Tools: 27.0.3, 28.0.3
        System Images: android-28 | Google APIs Intel x86 Atom, android-28 | Google APIs Intel x86 Atom_64, android-28 | Google Play Intel x86 Atom, android-28 | Google Play Intel x86 Atom_64
    IDEs:
      Android Studio: 3.3 AI-182.5107.16.33.5264788
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.5.0 => 16.5.0
      react-native: https://github.com/expo/react-native/archive/sdk-32.0.0.tar.gz => 0.57.1
    npmGlobalPackages:
      react-native-cli: 2.0.1
```

6. If you're developing for React Native iOS, run the following command and paste the output below: `pod --version`

1.5.3

7. Please enable verbose logging for your app using `[MSAppCenter setLogLevel: MSLogLevelVerbose];` before your call to `[AppCenterReactNative register];` (or any other SDK registration). For Android, use `AppCenter.setLogLevel(android.util.Log.VERBOSE);` before your `SoLoader.init` call. Include the logs here:

```
2019-05-27 16:17:07.325 [info][tid:main][RNGestureHandlerManager.m:135] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x7fe5fef27de0; reactTag: 1; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001bcd5f0>; layer = <CALayer: 0x60000143a020>>
2019-05-27 16:18:28.875 [info][tid:com.facebook.react.ShadowQueue][RNGestureHandlerManager.m:169] [GESTURE HANDLER] Tearing down gesture handler registered for views {(
    <RCTRootContentView: 0x7fe5fef27de0; reactTag: 1; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001bcd5f0>; layer = <CALayer: 0x60000143a020>>
)}
2019-05-27 16:18:34.826 [info][tid:main][RNGestureHandlerManager.m:135] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x7fe5fed330e0; reactTag: 11; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a552f0>; layer = <CALayer: 0x60000143f720>>
2019-05-27 16:19:31.816 [info][tid:com.facebook.react.ShadowQueue][RNGestureHandlerManager.m:169] [GESTURE HANDLER] Tearing down gesture handler registered for views {(
    <RCTRootContentView: 0x7fe5fed330e0; reactTag: 11; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a552f0>; layer = <CALayer: 0x60000143f720>>
)}
2019-05-27 16:19:36.022 [info][tid:main][RNGestureHandlerManager.m:135] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x7fe5fec9ac50; reactTag: 21; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a6e850>; layer = <CALayer: 0x600001481260>>
2019-05-27 16:19:41.852 [info][tid:com.facebook.react.ShadowQueue][RNGestureHandlerManager.m:169] [GESTURE HANDLER] Tearing down gesture handler registered for views {(
    <RCTRootContentView: 0x7fe5fec9ac50; reactTag: 21; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a6e850>; layer = <CALayer: 0x600001481260>>
)}
2019-05-27 16:19:45.154 [info][tid:main][RNGestureHandlerManager.m:135] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x7fe5fec88490; reactTag: 31; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a29a40>; layer = <CALayer: 0x6000014c37a0>>
2019-05-27 16:19:59.303 [info][tid:com.facebook.react.ShadowQueue][RNGestureHandlerManager.m:169] [GESTURE HANDLER] Tearing down gesture handler registered for views {(
    <RCTRootContentView: 0x7fe5fec88490; reactTag: 31; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a29a40>; layer = <CALayer: 0x6000014c37a0>>
)}
2019-05-27 16:20:01.084 [info][tid:main][RNGestureHandlerManager.m:135] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x7fe5fed1ceb0; reactTag: 41; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a17ea0>; layer = <CALayer: 0x6000014008c0>>
2019-05-27 16:21:31.079 [info][tid:com.facebook.react.ShadowQueue][RNGestureHandlerManager.m:169] [GESTURE HANDLER] Tearing down gesture handler registered for views {(
    <RCTRootContentView: 0x7fe5fed1ceb0; reactTag: 41; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a17ea0>; layer = <CALayer: 0x6000014008c0>>
)}
2019-05-27 16:21:37.379 [info][tid:main][RNGestureHandlerManager.m:135] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x7fe5fed3d1d0; reactTag: 51; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a2a3a0>; layer = <CALayer: 0x600001451540>>
2019-05-27 16:21:41.215 [info][tid:com.facebook.react.ShadowQueue][RNGestureHandlerManager.m:169] [GESTURE HANDLER] Tearing down gesture handler registered for views {(
    <RCTRootContentView: 0x7fe5fed3d1d0; reactTag: 51; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a2a3a0>; layer = <CALayer: 0x600001451540>>
)}
2019-05-27 16:21:45.910 [info][tid:main][RNGestureHandlerManager.m:135] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x7fe5fec32250; reactTag: 61; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a1db60>; layer = <CALayer: 0x6000015b9f60>>
2019-05-27 16:21:49.310 [info][tid:com.facebook.react.ShadowQueue][RNGestureHandlerManager.m:169] [GESTURE HANDLER] Tearing down gesture handler registered for views {(
    <RCTRootContentView: 0x7fe5fec32250; reactTag: 61; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a1db60>; layer = <CALayer: 0x6000015b9f60>>
)}
2019-05-27 16:21:52.356 [info][tid:main][RNGestureHandlerManager.m:135] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x7fe5fecc2d30; reactTag: 71; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a42c40>; layer = <CALayer: 0x600001580be0>>
2019-05-27 16:22:08.291 [info][tid:com.facebook.react.ShadowQueue][RNGestureHandlerManager.m:169] [GESTURE HANDLER] Tearing down gesture handler registered for views {(
    <RCTRootContentView: 0x7fe5fecc2d30; reactTag: 71; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a42c40>; layer = <CALayer: 0x600001580be0>>
)}
2019-05-27 16:22:11.749 [info][tid:main][RNGestureHandlerManager.m:135] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x7fe5fee66d90; reactTag: 81; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001ab43c0>; layer = <CALayer: 0x6000014460a0>>
2019-05-27 16:22:18.448 [info][tid:com.facebook.react.ShadowQueue][RNGestureHandlerManager.m:169] [GESTURE HANDLER] Tearing down gesture handler registered for views {(
    <RCTRootContentView: 0x7fe5fee66d90; reactTag: 81; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001ab43c0>; layer = <CALayer: 0x6000014460a0>>
)}
2019-05-27 16:22:26.481 [info][tid:main][RNGestureHandlerManager.m:135] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x7fe5fed3d1d0; reactTag: 101; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a22f10>; layer = <CALayer: 0x600001444f60>>
2019-05-27 16:22:30.824 [info][tid:com.facebook.react.ShadowQueue][RNGestureHandlerManager.m:169] [GESTURE HANDLER] Tearing down gesture handler registered for views {(
    <RCTRootContentView: 0x7fe5fed3d1d0; reactTag: 101; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001a22f10>; layer = <CALayer: 0x600001444f60>>
)}
2019-05-27 16:22:32.308 [info][tid:main][RNGestureHandlerManager.m:135] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x7fe5fec7a530; reactTag: 111; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600001ae5c20>; layer = <CALayer: 0x60000140e760>>
(lldb) 
```

Even though i have enabled the AppCenter to use VERBOSE logging, I don't seem to see any logs pertaining to AppCenter. I wonder why.

Let me know if I can get you guys more information.

EDIT: I actually have managed to get the Analytics to show up in the dashboard. It seems like the secret key is not being registered even though i have configured it in my AppCenter-Config.plist

If I add the following imports
```
@import AppCenter;
@import AppCenterAnalytics;
@import AppCenterCrashes;
```

and add the following inside `didFinishLaunchingWithOptions` in `AppDelegate.m`

```
[MSAppCenter setLogLevel: MSLogLevelVerbose];
[MSAppCenter configureWithAppSecret:(@"secretKey")];
```

Then the Analytics work and even tracking of events work.

Ideas?

**Comment by annakocheshkova** [comment]
@zanechua please share your `AppCenter-Config.plist` (put placeholder where your app secret is).
Is this file linked to the XCode project and where is it located?

**Comment by zanechua** [comment]
The file is located in `root/ios/appname/AppCenter-Config.plist`

The files contents are the following:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>AppSecret</key>
    <string>secretKey</string>
  </dict>
</plist>
```

The file was generated by react-native link and pretty sure the docs say to just replace the {APP_SECRET} value

**Comment by zanechua** [comment]
Ah. I have found the issue. I was digging into how the AppCenter-Config.plist was being picked up by the XCode project and realised that inside my project workspace, application, Build Phases under Copy Bundle Resources, AppCenter-Config.plist was not included. 

What hinted me to that was when I was searching this SDK repo, the AppCenter-Config.plist appears in the [project.pbxproj](https://github.com/microsoft/appcenter-sdk-react-native/blob/1cfb0432ef4ab94f1943c77c130811bddec9fcc2/TestApp34/ios/TestApp34.xcodeproj/project.pbxproj) of the various test applications you guys have in here.

Could this be a appcenter-link-scripts issue as I believe that package is responsible for making sure the AppCenter-Config.plist file is registered in the project.pbxproj? If that's not the case, perhaps the documentation needs to be updated so that other users won't face this issue.

**Comment by Jamminroot** [comment]
@zanechua We couldn't actually reproduce the problem. 
As for linking script - we also couldn't find any errors with it and it worked as expected.
Would you like us to look into it a bit deeper, more in-depth repro steps could help us out.
Closing thread, but feel free to reopen

**Comment by annakocheshkova** [comment]
Yes @zanechua as @Jamminroot mentioned we didn't face this issue when running link script, perhaps yours failed with some kind of error and then it didn't link the config file.
As for the manual integration - you are responsible for creating and linking this file to the project. Maybe we could make this more clear in the documentation, what do you think?

**Comment by zanechua** [comment]
Just did a quick test with a clean expo project after ejection and you're right. I can't seem to reproduce the problem either. Guess this will be here for reference in-case anyone else manages to reproduce the unique situation where the config file didn't get added into the bundle resources..

@annakocheshkova Yes I think this is a good idea to add in the docs for developers to do a double check to make sure that their AppConfig-Center.plist file is included in their project. Perhaps in the FAQ section? I didn't do manual integration and I did the react-native link way but that didn't seem to work for me at least. For someone coming from a web background, I'm not particularly familiar with where the file should be added/registered in an XCode project.

**Comment by annakocheshkova** [comment]
@zanechua cool, I've added a note to our docs.
About your workflow, I'm not sure I follow: so the config file was already created - when exactly? And if it was added previously by `react-native link`, then it should already be linked to the project, right?

**Comment by zanechua** [comment]
Okay. To elaborate a little more.

The expo toolchain allows us to just write react-native code without worrying about the ios or android builds. However we required some native features which requires us to eject the app and that generates out the ios and android folders converting it into a slightly more traditional react-native application where you can manage and add native plugins specific for ios or android.

In my branch, I had an expo build before ejection and I ejected my version which generated the ios and android folders. But my integration was only for AppCenter and I did not write any native code but my colleague did so we decided to use his version where AppCenter was not integrated. I had already run `react-native link` on my end which caused `AppCenter-Config.plist` to be created and added to the `project.pbxproj` file. I then ran `git reset --hard origin/branch` which resetted my build but didn't delete the `AppCenter-Config.plist` then I checked out his branch that had the ios and android folders. Then I ran `react-native link` again.

Basically the `AppCenter-Config.plist` is only added to `project.pbxproj` on the assumption that the file does not exist. If it exists, it assumes that the configuration for the file is inside `project.pbxproj`. Perhaps an extra check can be introduced to check if the file is added into `project.pbxproj` and if it isn't, add it in even if the `AppCenter-Config.plist` exists.

**Comment by annakocheshkova** [comment]
@zanechua seems like a rare use-case, although definitely possible. 
I will confirm with the team if this is something we would implement. The current impression is that this is not just a simple check but more like a chunk of code with complex parsing logic, which will possibly complicate our link script a lot and make it more bug-prone. Altough if there's a simple way to make this check, of course, we will.

**Comment by zanechua** [comment]
Yeah. No worries. I think a note in the docs would be good for people to double check that it's added and they wouldn't have to spend so much time figuring out what exactly is going on like I did.

Thanks for the super response time. Much appreciated.  🙂

**Comment by annakocheshkova** [comment]
Hey @zanechua just wanted to let you know that we updated the [React Native getting started docs](https://docs.microsoft.com/en-us/appcenter/sdk/getting-started/react-native#31-integrate-the-sdk-automatically) (point 3.1 - 3, 3.2 - 4, 3.3 - 12). Please let us know if they seem clear to you!

**Comment by zanechua** [comment]
Looks good @annakocheshkova 

Definitely more explicit now!
